### ISSUE TITLE
RowDecoder.ordered and RowDecoder.decoder don't produce Serializable decoders

### ISSUE THREAD

**Issue by aroberts** [nrinaudo/kantan.csv#81]
I'm running into an issue when using `RowDecoder.ordered`, and I replicated it with `RowDecoder.decoder`: my large external framework complains that the decoder isn't serializable, pointing a vague finger at the `GeneratedRowDecoder` trait. Notably, `RowDecoder.from` DOES produce serializable decoders, presumably because the non-serializable trait somehow isn't mixed in.

Here's the (java) code I'm using to test "serializability" (from [Flink](http://github.com/apache/flink))
```java
public static byte[] serializeObject(Object o) throws IOException {
	try (ByteArrayOutputStream baos = new ByteArrayOutputStream();
			ObjectOutputStream oos = new ObjectOutputStream(baos)) {
		oos.writeObject(o);
		oos.flush();
		return baos.toByteArray();
	}
}
```

**Comment by nrinaudo** [comment]
Can you provide more information about what you mean by "is or isn't serializable" ?
Do you mean:
* it extends `java.io.Serializable`
* it extends `scala.Serializable`
* you tried serialising it and it failed
?

**Comment by nrinaudo** [comment]
Right. This is due to the fact that all `GeneratedRowDecoder` decoders refer to `GeneratedRowDecoder.decode`, which breaks serialisation. The fix is fairly simple - move this function to, say, `kantan.csv`'s package object (this has been confirmed to work).

I will address this soon(ish), but need to take the time to write proper serialisation / deserialisation tests first - it's a sneaky bug that I'm sure impacts lots of other decoders (I believe java8, joda, anything with `Option` in it...)

**Comment by nrinaudo** [comment]
Fixed in 0.1.19. Closing this for now, but feel free to reopen it if you feel the fix doesn't meet your requirements.
